src/python/gudhi/periodic_cubical_complex.pyx

```diff
@@ -60,16 +60,41 @@ cdef class PeriodicCubicalComplex:
 
         Or
 
+        :param numpy_array: Filtration values in a d-array.
+        :type numpy_array: numpy ndarray
+        :param periodic_dimensions: A list of top dimensional cells periodicity value.
+        :type periodic_dimensions: list of boolean
+
+        Or
+
         :param perseus_file: A Perseus-style file name.
         :type perseus_file: string
         """
 
     # The real cython constructor
     def __cinit__(self, dimensions=None, top_dimensional_cells=None,
-                  periodic_dimensions=None, perseus_file=''):
-        if (dimensions is not None) and (top_dimensional_cells is not None) and (periodic_dimensions is not None) and (perseus_file == ''):
-            self.thisptr = new Periodic_cubical_complex_base_interface(dimensions, top_dimensional_cells, periodic_dimensions)
-        elif (dimensions is None) and (top_dimensional_cells is None) and (periodic_dimensions is None) and (perseus_file != ''):
+                  numpy_array=None, periodic_dimensions=None,
+                  perseus_file=''):
+        if ((dimensions is not None) and (top_dimensional_cells is not None)
+            and (numpy_array is None) and (periodic_dimensions is not None)
+            and (perseus_file == '')):
+            self.thisptr = new Periodic_cubical_complex_base_interface(dimensions,
+                                                                       top_dimensional_cells,
+                                                                       periodic_dimensions)
+        elif ((dimensions is None) and (top_dimensional_cells is None)
+            and (numpy_array is not None) and (periodic_dimensions is not None)
+            and (perseus_file == '')):
+            if isinstance(numpy_array, np.ndarray):
+                dimensions = list(numpy_array.shape)
+                top_dimensional_cells = numpy_array.flatten().tolist()
```

Comment:
1. flatten always makes a copy, while some variants make a copy only if needed (flat, ravel, etc)
2. do we really need to convert to a list? I hoped cython would know how to convert a numpy array to a vector, just like it knows how to convert a list.

Reply:
You are right. Fixed on 32a9c5e4df26e59f5f376e54c05c7bec92012cff
I used [numpy.ravel('C')](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ravel.html#numpy.ravel). I hope this is the good candidate.